aiohttp/web_protocol.py

```diff
@@ -633,26 +633,30 @@ async def start(self) -> None:
 
             except asyncio.CancelledError:
                 self.log_debug("Ignored premature client disconnection")
+                self.force_close()
                 raise
             except Exception as exc:
                 self.log_exception("Unhandled exception", exc_info=exc)
                 self.force_close()
+            except BaseException:
+                self.force_close()
+                raise
+            else:
+                if self._keepalive and not self._close:
```

Comment:
Could `self._force_close` get set somewhere else outside of this loop.. do we still need to check it?